pkg/epp/scheduling/plugins/scorer/prefix_store.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scorer
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+	"sync"
+	"time"
+
+	"github.com/cespare/xxhash/v2"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const (
+	// defaultMaxCacheSize sets the maximum number of blocks the LRU cache can store.
+	defaultMaxCacheSize = 500000
+	// defaultBlockSize defines how many runes each block contains in the prefix cache.
+	defaultBlockSize = 256
+	// defaultMaxBlockCacheSize sets the maximum number of pods a block can store.
+	defaultMaxBlockCacheSize = 100
+)
+
+// PrefixStoreConfig contains initialization configuration for PrefixStore.
+type PrefixStoreConfig struct {
+	// CacheSize sets the maximum number of blocks the LRU cache can store.
+	CacheSize int
+	// BlockSize defines how many runes each block contains in the prefix cache.
+	BlockSize int
+	// BlockCacheSize sets the maximum number of pods a block can store.
+	BlockCacheSize int
+}
+
+// DefaultPrefixStoreConfig returns an PrefixStoreConfig instance with default
+// configuration.
+func DefaultPrefixStoreConfig() *PrefixStoreConfig {
+	return &PrefixStoreConfig{
+		CacheSize:      defaultMaxCacheSize,
+		BlockSize:      defaultBlockSize,
+		BlockCacheSize: defaultMaxBlockCacheSize,
+	}
+}
+
+// block holds the tokens contained in the block.
+type block struct {
+	Pods *lru.Cache[types.NamespacedName, time.Time] //TODO: implement Pod eviction based on staleness
+}
+
+// PrefixStore is an in-memory prefix-to-block cache with xxhash keys and LRU
```

Comment:
```suggestion
// PrefixStore is an in-memory prefix-to-block cache with hash keys and LRU
```
Why commit to `xxhash` at this point?

Reply:
Any reason not to? I saw it has nice performance, and it was used in a reference implementation I saw.